### PR TITLE
Make gen4 the default planner for release 14

### PR DESCRIPTION
**Pull request by systay**
## Description
Release 14 is still a long way away, but we might as well do this change now to flush out any bugs.

## Checklist
- [x] Should this PR be backported? ** NO **
- [x] Tests were added or are not required
- [ ] Documentation was added or is not required

### CODE REVIEW

**Review by GuptaManan100 — APPROVED** [review]

**Comment by GuptaManan100** [comment]
🤞

File: go/vt/vtgate/planbuilder/builder.go [hunk 1]
```diff
@@ -111,9 +111,11 @@ func getConfiguredPlanner(vschema plancontext.VSchema, v3planner func(string) se
...
 		}
 		return fp.plan, nil
-	default:
-		// default is v3 plan
+	case V3:
 		return v3planner(query), nil
+	default:
+		// default is gen4 plan
+		return gen4Planner(query, Gen4), nil
 	}
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/builder.go:119]: nit: remove case Gen4 from above.

Comment (systay) [reply go/vt/vtgate/planbuilder/builder.go:119]: the gen4 case is still needed. There are variations of gen4 (left2right and greedy only) that are not handled by the default case

**Comment by frouioui** [comment]
With Gen4 being the new default planner, `TestExplain` started failing consistently on the `selectsharded` queries. I have detailed the issue in #9854 and proposed a fix for it in #9855.

**Comment by harshit-gangal** [comment]
We should mark the flag for deprecation as well.

**Pull request merged**